internal/http/config.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package http
+
+import "github.com/open-telemetry/opentelemetry-collector/receiver"
+
+// TODO: Pull in other utilities from
+// https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/common/httpclient/http.go.
+// HTTPConfig holds config options related to HTTP(S)
+type HTTPConfig struct {
+	// Whether not TLS is enabled
+	TLSEnabled bool      `mapstructure:"tls_enabled"`
+	TLSConfig  TLSConfig `mapstructure:"tls_config"`
+}
+
+// TLSConfig holds common TLS config options
+type TLSConfig struct {
+	receiver.TLSCredentials `mapstructure:",squash"`
```

Comment:
@tigrannajaryan @bogdandrutu @pjanotti I think this is what we agreed to but it's not clear to me what value this provides. I feel like it might be better to just make a clean break from `TLSCredentials`. When `TLSConfig` in contrib is ready we can deprecate `TLSCredentials` and migrate everything to `TLSConfig`.

Reply:
But I'm also ok with this if there's no clear consensus and can revisit later.

Reply:
>I feel like it might be better to just make a clean break from TLSCredentials

@jrcamp I agree, I don't see the value of coupling http client config with `receiver.TLSCredentials`. I don't remember if I suggested this approach, if I did then sorry I was not clear since I did not mean it.

Reply:
If everyone agrees, I can revert the last commit to decouple the two structs. Or if this is not the approach we want to take, please let me know what the right approach is for the short term since we have other components currently being worked on that benefit from this update. @tigrannajaryan @pjanotti @bogdandrutu

Reply:
I'm going the issue to converge the tls credentials in core because I need to implement mTLS between a few exporter -> receivers  https://github.com/open-telemetry/opentelemetry-collector/issues/933